### PR TITLE
Add basic guide

### LINKED ISSUE #13: Create documentation
Both java classes and user guide. Having something like protostar documentation would be nice.

### PR DESCRIPTION
**Pull request by cptartur**
This PR adds a basic guide to the library. Closes #13.

### CODE REVIEW

File: lib/lib.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    }
+}
+```
+
+## Example usages in Kotlin
+
+```kotlin
+fun main() {
+    // Create a provider for interacting with StarkNet
+    val provider = GatewayProvider("feeder_gateway_url", "gateway_url", StarknetChainId.TESTNET)
```
Comment (Solpatium) [inline lib/lib.md:37]: Don't we provide those values as some constants? We probably should do that or expose provider configured for testnet already.

File: lib/lib.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        var provider = new GatewayProvider("feeder_gateway_url", "gateway_url", StarknetChainId.TESTNET);
+
+        // Create an account interface
+        Felt accountAddress = new Felt(45234235672347L);
+        Felt privateKey = new Felt(5326326273453L);
+        var account = new StandardAccount(provider, accountAddress, privateKey);
+
+        // Make an asynchronous request
+        Felt contractAddress = new Felt(42362362436L);
+        Felt storageKey = new Felt(13241253414L);
```
Comment (Solpatium) [inline lib/lib.md:68]: This is not realistic, please use `fromHex`.

**Review by Solpatium — CHANGES_REQUESTED** [review]

File: lib/lib.md (hunk 1 above)
Comment (cptartur) [reply lib/lib.md:37]: I don't think we have these declared as constants at the moment.

Comment (Solpatium) [reply lib/lib.md:37]: Then we should do something about it

Comment (cptartur) [reply lib/lib.md:37]: These has been added along with gateway client factory. Examples will need to be updated to reflect these changes.

Comment (Solpatium) [reply lib/lib.md:37]: Where is the factory?

Comment (cptartur) [reply lib/lib.md:37]: In the GatewayProvider companion object

Comment (Solpatium) [reply lib/lib.md:37]: Yes, please update the docs to use makeTestnetClient

File: lib/lib.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
+# Module lib
+
+StarkNet.kt is a library allowing for easy interaction with StarkNet gateway and nodes, including
+querying starknet state, executing transactions and deploying contracts.
+
+Although written in Kotlin, StarkNet.kt has been created with compatibility with Java in mind.
+
+## Example usages in Java
```
Comment (Solpatium) [inline lib/lib.md:8]: usage

Comment (Solpatium) [inline lib/lib.md:8]: We can structure it a little bit better:
- Asynchronous requests
  - java
  - kotlin
- Synchronous requests
  - java
  - kotlin

File: lib/lib.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+        response.thenAccept(System.out::println);
+    }
+}
+```
+
+### In Kotlin
+
+It is also possible to make asynchronous requests. `Request.sendAsync()` returs a `CompletableFuture`
+that can be than handled in preferred way.
```
Comment (Solpatium) [inline lib/lib.md:106]: Move this at the beginning of asynchronous section.

**Review by Solpatium — APPROVED** [review]

**Pull request merged**